pyperformance/data-files/benchmarks/bm_btree/run_benchmark.py

```diff
@@ -0,0 +1,418 @@
+                            + node.items
+                        )
+                        if not node.is_leaf():
+                            node.nodes = lower_sibling.nodes + node.nodes
+                        del self.items[p - 1]
+                        del self.nodes[p - 1]
+                    else:
+                        # Case 3b2: Merge with upper_sibling
+                        node.items = (
+                            node.items + [self.items[p]] + upper_sibling.items
+                        )
+                        if not node.is_leaf():
+                            node.nodes = node.nodes + upper_sibling.nodes
+                        del self.items[p]
+                        del self.nodes[p + 1]
+                assert is_big(node)
+                node.delete(key)
+            if not self.items:
+                # This can happen when self is the root node.
+                self.items = self.nodes[0].items
+                self.nodes = self.nodes[0].nodes
+
+
+class BTree:
+    """
+    Instance attributes:
+      root: BNode
+    """
+
+    __slots__ = ['root']
+
+    def __init__(self):
+        self.root = BNode()
+
+    def __bool__(self):
+        return bool(self.root.items)
+
+    def iteritems(self):
+        yield from self.root
+
+    def iterkeys(self):
+        for item in self.root:
+            yield item[0]
+
+    def __iter__(self):
+        yield from self.iterkeys()
+
+    def __contains__(self, key):
+        return self.root.search(key) is not None
+
+    def __setitem__(self, key, value):
+        self.add(key, value)
+
+    def __getitem__(self, key):
+        item = self.root.search(key)
+        if item is None:
+            raise KeyError(key)
+        return item[1]
+
+    def __delitem__(self, key):
+        self.root.delete(key)
+
+    def get(self, key, default=None):
+        """(key:anything, default:anything=None) -> anything"""
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def add(self, key, value=True):
+        """(key:anything, value:anything=True)
+        Make self[key] == val.
+        """
+        if self.root.is_full():
+            # replace and split.
+            node = self.root.__class__()
+            node.nodes = [self.root]
+            node.split_child(0, node.nodes[0])
+            self.root = node
+        self.root.insert_item((key, value))
+
+    def __len__(self):
+        """() -> int
+        Compute and return the total number of items."""
+        return self.root.get_count()
+
+
+class Record:
+    def __init__(self, a, b, c, d, e, f):
+        self.a = a
+        self.b = b
+        self.c = c
+        self.d = d
+        self.e = e
+        self.f = f
+
+
+def make_records(num_nodes):
+    rnd = random.Random(RANDOM_SEED)
+    for node_id in range(num_nodes):
+        a = node_id
+        b = f'node {node_id}'
+        c = rnd.randbytes(node_id % 100)
```

Comment:
Introduced in Python 3.9, while:

https://github.com/python/pyperformance/pull/381/files#diff-5ac73097bc69dcf4f2f194138de6c0445c44b59541706d887b5902e8e54641f3R3